### PR TITLE
fix(form/inputs): only scroll and select focusPath when needed

### PR DESCRIPTION
**Pull request by skogsmaskin**
### Description

The PT-input is doing focusPath tracking in some situations it should not, which can cause issues when editing embedded objects inside through the dialogs.

### What to review

* That we don't make a new selection (or scrollIntoView) in the editor when an object is being edited or multiple times.
* That studio links pointing to content inside a PT-input is properly selected, focused and scrolled to
* That the selection is correct when toggling fullscreen in the PT-input
* That clicking on top navigation validation items for the PT-input is properly selected, focused and scrolled to

### Notes for release

Doesn't need to be mentioned, it's a unreleased regression from https://github.com/sanity-io/sanity/pull/4466

### CODE REVIEW

File: packages/sanity/src/core/form/inputs/PortableText/hooks/useTrackFocusPath.tsx [hunk 1]
```diff
@@ -27,6 +27,13 @@ export function useTrackFocusPath(props: Props): void {
       .filter((item) => item.member.open)
       .sort((a, b) => b.member.item.path.length - a.member.item.path.length)[0]
     if (memberItem && memberItem.elementRef?.current) {
+      // Don't do anything if we have already "closed" a textBlock (see below) or have a object open for editing.
+      const isTextBlockClosed = memberItem.kind === 'textBlock' && !memberItem.member.open
+      const isObjectOpen = memberItem.kind !== 'textBlock' && memberItem.member.open
```
Comment (skogsmaskin) [inline packages/sanity/src/core/form/inputs/PortableText/hooks/useTrackFocusPath.tsx:32]: Will this be true with the case where you only have an object "selected" in the editor, but not having focus inside?

Ref. what we talked about offline @bjoerge

**Review by skogsmaskin — COMMENTED** [review]
I'm a bit unsure how this will act  if we decide to change the formState `open` property for the member.

**Comment by skogsmaskin** [comment]
Force pushed a new solution that probably works better.
Just do it once per opened `memberItem`.

**Comment by skogsmaskin** [comment]
I have identified another regression here too, when creating new annotations and inline objects, the selection screwed up from the focus tracking.

**Comment by skogsmaskin** [comment]
Force pushed a new approach that doesn't interfere with creating links and inline objects.

File: .gitignore [hunk 2]
```diff
@@ -71,3 +71,7 @@ etc
 /test/e2e/report
 /test/e2e/results
 /test/e2e/state
+
+# Temporary files made by yarn when running canary releases
+.yarn
+.yarnrc
```
Comment (RitaDias) [inline .gitignore:77]: Do these also ignore the .cjs files that are appearing bellow on the files changes? 👇

**Review by RitaDias — APPROVED** [review]
After testing it around, it looks good :)

**Review by bjoerge — APPROVED** [review]
LGTM!

**Comment by skogsmaskin** [comment]
These changes will be part of another PR that @bjoerge will merge in later on.

Closing this for now.

**Comment by skogsmaskin** [comment]
This will be superseded by Superseded by https://github.com/sanity-io/sanity/pull/4524

**Pull request closed without merging**